HDFDataset.py

```diff
@@ -84,34 +85,38 @@ def add_file(self, filename):
         self.timestamps = fin[attr_times][...]
       else:
         self.timestamps = numpy.concatenate([self.timestamps, fin[attr_times][...]], axis=0)
-    seq_lengths = fin[attr_seqLengths][...]
     if 'targets' in fin:
       self.target_keys = sorted(fin['targets/labels'].keys())
     else:
       self.target_keys = ['classes']
 
+    seq_lengths = fin[attr_seqLengths][...]
     if len(seq_lengths.shape) == 1:
       seq_lengths = numpy.array(zip(*[seq_lengths.tolist() for i in range(len(self.target_keys)+1)]))
 
-    if len(self._seq_lengths) == 0:
-      self._seq_lengths = numpy.array(seq_lengths)
-    else:
-      self._seq_lengths = numpy.concatenate((self._seq_lengths, seq_lengths), axis=0)
-    if not self._seq_start:
-      self._seq_start = [numpy.zeros((seq_lengths.shape[1],), 'int64')]
     seq_start = numpy.zeros((seq_lengths.shape[0] + 1, seq_lengths.shape[1]), dtype="int64")
     numpy.cumsum(seq_lengths, axis=0, dtype="int64", out=seq_start[1:])
-    self._tags += fin["seqTags"][...].tolist()
-    self.file_seq_start.append(seq_start)
```

Comment:
Btw, this is mostly just reordering. I wanted to keep the time where seq_lengths is in memory as short as possible.